### ISSUE TITLE
SdkController line 27: NullPointerException crash

### ISSUE THREAD

**Issue by rygelouv** [onfido/onfido-android-sdk#229]
#### Description
Crash happening on `OnfidoActivity`

#### Version info:
- `onfido-capture-sdk` version: 15.2.0
- Android API version(s): Android 9 - 13

- `compileSdkVersion`: 33
- `targetSdkVersion`: 33
- `minSdkVersion`: 24
- Support library version (if applicable):

#### Steps to reproduce:
Unkown

#### Issue severity information (number of users/crashes if applicable):
This issue has 7 crash events affecting 7 users in 7 days.
#### Additional LogCat output:
```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mychime.waveremit.app/com.onfido.android.sdk.capture.ui.OnfidoActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4035)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4201)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2438)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8663)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
```
#### Other useful info:
From Crashlytics: _"This crash event happened in the first second of the user's session."_

**Comment by gurpreet-onfido** [comment]
Hello @rygelouv 

Greetings for the day!

As you have mentioned that there are no steps to reproduce the reported issue, it becomes a bit difficult for us to advise what can cause this problem. In order to prioritise the resolution of this issue on our side, could you please help me with the following details: 

1. Is this issue also happening on the latest SDK release that is 15.4.1?
2. What is the impact of this issue ? We are trying to understand how many % of the total traffic face this issue? Were they able to complete the flow on another try? Example: 800 of XYZ users have run into this issue in the past week, which is 10% of all their Onfido traffic).
3. Consequence of the issue (ie. bad end-user experience, can't complete flow, etc).

This will help us to raise this internally and we will be in a better position to advise the ETA.

Thank you so much for your understanding & cooperation.

Kind regards
Gurpreet Singh

**Comment by rygelouv** [comment]
Hello @gurpreet-onfido 

Thanks for you answer.

1 - We released a version last week which has the latest onfido SDK version (15.4.1) and for the last 7 days on that specific version, Crashlytics is already reporting **90 crash events affecting 69 users**

2 - It's hard to tell even for us what is the % of total traffic. Our app is distributed in many countries and not all of them have to go through Onfio. Also we start Onfido at various points in the user journey for different reasons. So it's quite difficult to tell. Since this is an early crash happening in the first second of the user session, Crashlytics is not having enough time to get the attached the user ID. With the user ID we could have checked if those users have completed Onfido on another try.

3 - As said in "2.", this is an early crash happening in the first second of the user session which means if the issue is persistant, the some user are not even able to run the app. I mean not only Onfido flow but literally the entire app.

I'll keep looking if I can get an ID from one of our user and check if they were able to still complete the flow.
Please do ask for more details if it can help, I'll keep providing those I'm able to provide.

Thanks

**Comment by dbottillo** [comment]
> Hello @rygelouv
> 
> Greetings for the day!
> 
> As you have mentioned that there are no steps to reproduce the reported issue, it becomes a bit difficult for us to advise what can cause this problem. In order to prioritise the resolution of this issue on our side, could you please help me with the following details:
> 
> 1. Is this issue also happening on the latest SDK release that is 15.4.1?
> 2. What is the impact of this issue ? We are trying to understand how many % of the total traffic face this issue? Were they able to complete the flow on another try? Example: 800 of XYZ users have run into this issue in the past week, which is 10% of all their Onfido traffic).
> 3. Consequence of the issue (ie. bad end-user experience, can't complete flow, etc).
> 
> This will help us to raise this internally and we will be in a better position to advise the ETA.
> 
> Thank you so much for your understanding & cooperation.
> 
> Kind regards Gurpreet Singh

Hi, we are experiencing the same issue on our app as well and It seems to be happening on `15.4.1` as well. For us it's the highest crash that we have in the app and happening roughly to 10% of users and yes it's a fatal crash so they can't complete the flow. Is there any update on this issue?

**Comment by gurpreet-onfido** [comment]
Hello @rygelouv and @dbottillo 

Thank you for sharing this information with us. I have passed it on to our engineering team for the review. As soon as an update is available with me, I will update this thread. 

Stay tuned!

Kind regards
Gurpreet Singh

**Comment by gurpreet-onfido** [comment]
Hello @rygelouv 

Greetings for the day!

Our SDK team is working on this issue and they have mentioned that the shared stack trace is incomplete and doesn’t provide much valuable debugging info. We need the full stack trace to be able to investigate this issue and identify potential causes. 

Could you please provide the complete stack trace. 

Also, we need the SDK initialization code to see if there is something particular about it that can contribute to the crash. Could you please provide us with the same. 

Looking forward to your response.

Kind regards
Gurpreet Singh

**Comment by rygelouv** [comment]
@gurpreet-onfido 
Looking at our crashlytics it seems indeed that some logs are missing. Another block of strack trace shows this:

```
Caused by java.lang.NullPointerException:
       at com.onfido.android.sdk.capture.common.SdkController.getSdkComponent(SourceFile:27)
       at com.onfido.android.sdk.capture.common.SdkController.getSdkComponent$default(SourceFile:11)
       at com.onfido.android.sdk.capture.ui.OnfidoActivity.onCreate(SourceFile:16)
       at android.app.Activity.performCreate(Activity.java:8207)
       at android.app.Activity.performCreate(Activity.java:8191)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3819)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4022)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2336)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8653)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

```
This shows that `OnfidoActivity.onCreate()` is being called first then it calls `SdkController.getSdkComponent` then it crashes.

**Comment by rygelouv** [comment]
@gurpreet-onfido 
Maybe also worth sharing that a separate crash also happening that is strangely similar to this but instead of crashing on `OnfidoActivity`, it crashes on `CaptureActivity`. 
Here goes the stack trace:

```
Fatal Exception: java.lang.RuntimeException
Unable to start activity ComponentInfo{com.mychime.waveremit.app/com.onfido.android.sdk.capture.ui.camera.CaptureActivity}: android.view.InflateException: Binary XML file line #113 in com.mychime.waveremit.app:layout/onfido_activity_capture: Binary XML file line #113 in com.mychime.waveremit.app:layout/onfido_activity_capture: Error inflating class com.onfido.android.sdk.capture.validation.CaptureValidationBubble
```

```
Fatal Exception: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.mychime.waveremit.app/com.onfido.android.sdk.capture.ui.camera.CaptureActivity}: android.view.InflateException: Binary XML file line #113 in com.mychime.waveremit.app:layout/onfido_activity_capture: Binary XML file line #113 in com.mychime.waveremit.app:layout/onfido_activity_capture: Error inflating class com.onfido.android.sdk.capture.validation.CaptureValidationBubble
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4166)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4312)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8741)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

`Caused by android.view.InflateException: Binary XML file line #113 in com.mychime.waveremit.app:layout/onfido_activity_capture: Binary XML file line #113 in com.mychime.waveremit.app:layout/onfido_activity_capture: Error inflating class com.onfido.android.sdk.capture.validation.CaptureValidationBubble`

```
Caused by java.lang.reflect.InvocationTargetException:
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at android.view.LayoutInflater.createView(LayoutInflater.java:876)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1028)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:983)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:1145)
       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1106)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:692)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:544)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:491)
       at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:775)
       at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:197)
       at com.onfido.android.sdk.capture.ui.camera.CaptureActivity.onCreate(SourceFile:6)
       at android.app.Activity.performCreate(Activity.java:8578)
       at android.app.Activity.performCreate(Activity.java:8557)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4147)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4312)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8741)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```

```
Caused by java.lang.NullPointerException:
       at com.onfido.android.sdk.capture.common.SdkController.getSdkComponent(SourceFile:28)
       at com.onfido.android.sdk.capture.common.SdkController.getSdkComponent$default(SourceFile:12)
       at com.onfido.android.sdk.capture.validation.CaptureValidationBubble.<init>(SourceFile:4)
       at com.onfido.android.sdk.capture.validation.CaptureValidationBubble.<init>(SourceFile:10)
       at com.onfido.android.sdk.capture.validation.CaptureValidationBubble.<init>(SourceFile:2)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at android.view.LayoutInflater.createView(LayoutInflater.java:876)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:1028)
       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:983)
       at android.view.LayoutInflater.rInflate(LayoutInflater.java:1145)
       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1106)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:692)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:544)
       at android.view.LayoutInflater.inflate(LayoutInflater.java:491)
       at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:775)
       at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:197)
       at com.onfido.android.sdk.capture.ui.camera.CaptureActivity.onCreate(SourceFile:6)
       at android.app.Activity.performCreate(Activity.java:8578)
       at android.app.Activity.performCreate(Activity.java:8557)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1384)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4147)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4312)
       at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
       at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2571)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:226)
       at android.os.Looper.loop(Looper.java:313)
       at android.app.ActivityThread.main(ActivityThread.java:8741)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1067)
```
These two issue certainly have the same cause.

**Comment by gurpreet-onfido** [comment]
Hello @rygelouv Thank you for the detailed information. I have passed it to our engineering team for their review. 

We will get back to you on this as soon as more information is available with us. 

Kind regards
Gurpreet Singh

**Comment by gurpreet-onfido** [comment]
Hello everyone!

We are able to reproduce the reported at our end as well and we have prepared a fix for the same.

We will let you know once the new SDK release containing a fix is rolled out.

Please stay tuned!

Kind regards
Gurpreet Singh

**Comment by gurpreet-onfido** [comment]
Hello there!

Greetings for the day!

We have released a new version 16.1.0 of our Android SDK containing the fix. Could you please upgrade the SDK version at your end and let us know how it goes. 

We will wait for your feedback.

Kind regards
Gurpreet Singh

**Comment by rygelouv** [comment]
Hi @gurpreet-onfido 
Thanks for the heads
We'll work on the update and let you know
